src/components/input/input.tsx

```diff
@@ -119,23 +129,21 @@ export class Input {
      */
     @Prop() type: TextFieldTypes = 'text';
 
-    @Event() wcsChange!: EventEmitter<InputChangeEventDetail>;
-
     private getValue(): string {
         return this.value || '';
     }
 
     @Watch('value')
     protected valueChanged() {
         this.wcsChange.emit({ value: this.value });
-        console.log(this.value);
     }
 
-    private onInput(ev: Event) {
+    private onInput = (ev: Event) => {
```

Comment:
Declaring methods as fat arrow functions results in them not using the same memory space as normal ones and is heavier.

Is there any reason it has been changed ?

Also, `KeyboardEvent` should be used instead.

Reply:
Yes, the method must send an event from a property in the input component, so it had to remain in its scope.

Reply:
https://stenciljs.com/docs/templating-jsx

The solution is to keep a normal function and provide the scope in the render function instead :

```jsx
onInput={(evt: KeyboardEvent) => this.onInput(evt)}
// or
onInput={this.onInput.bind(this)}
```